### PR TITLE
Docker direct scoring (batch predict)

### PR DESCRIPTION
**Pull request by zhiyongzachng**
*changes to batch_predict.py for docker-direct scoring
*to be merged to api-integration branch*

### CODE REVIEW

**Comment by zhiyongzachng** [comment]
Did necessary changes such that each_chunk (each data row being set to predict function) is converted to .json format to be passed to dictator client in predict function.

File: batch_predict.py [hunk 1]
```diff
@@ -36,9 +37,10 @@ def _create_local_path(self, remote_path, local_prefix):
         remote_filename = remote_path.rpartition('/')[2]
 
         local_dir = os.path.join(settings.DATATRON_ROOT_LOCATION, self.workspace_slug, local_prefix)
-
+        
+        #make a local directory if it does not exists
         if not os.path.exists(local_dir):
-            os.makedirs(local_dir, exist_ok=True)
+            os.makedirs(local_dir, exist_ok=True)
```
Comment (kevindatatron) [inline batch_predict.py:43]: Do these white-spaces you added serve any purpose? If not, please remove them, they appear multiple times throughout this PR.

Comment (zhiyongzachng) [reply batch_predict.py:43]: True, not sure why its occurring. Will take note and make changes

File: batch_predict.py [hunk 2]
```diff
@@ -2,10 +2,13 @@
...
 import time
 import logging
+import json
 import pandas as pd
+import ast
 from app.utils import hdfs_transfer as ht
-from app.ml_model import predictor
 from app.settings import settings
+import requests
+from datatron.common.discovery import DatatronDiscovery
```
Comment (aaraddhyabhatalkar) [inline batch_predict.py:11]: Why do you need Datatron discovery?

Comment (zhiyongzachng) [reply batch_predict.py:11]: I need to get api call to Dictator "batch_prediction" table to cross check the features with those present in the input dataset. @aaraddhyabhatalkar

**Pull request closed without merging**